logger.ts

```diff
@@ -109,5 +115,40 @@ export class Logger implements ILogger {
 
 		return printMsgFuture;
 	}
+
+	private getPasswordEncodedArguments(args: string[]): string[] {
+		return _.map(args, a => {
+			if (typeof a !== 'string') {
+				return a;
+			}
+
+			var passwordMatch = this.passwordRegex.exec(a);
+			if (passwordMatch) {
+				var password = passwordMatch[1] || passwordMatch[3];
+				return this.getEncodedPassword(password, a);
+			}
+
+			if (this.encodeBody) {
+				var bodyMatch = this.requestBodyRegex.exec(a);
+				if (bodyMatch) {
+					return this.getEncodedPassword(bodyMatch[1], a);
+				}
+			}
+
+			for (var i = 0; i < this.encodeRequestPaths.length; ++i) {
+				var path = this.encodeRequestPaths[i];
+				if (a.indexOf('path') > -1) {
+					this.encodeBody = a.indexOf(path) > -1;
+					break;
+				}
+			}
+
+			return a;
+		});
+	}
+
+	private getEncodedPassword(password: string, originalString: string) {
```

Comment:
The name of this method does not reflect its function. It does not get the password, it replaces it in the string. Perhaps something similar to ```hidePassword```. Note, the password is not encoded, it is removed.